core/src/test/java/org/elasticsearch/action/search/SearchPhaseExecutionExceptionTests.java

```diff
@@ -0,0 +1,178 @@
+                                "\"node\":\"node_2\"," +
+                                "\"reason\":{" +
+                                            "\"type\":\"index_shard_closed_exception\"," +
+                                            "\"reason\":\"CurrentState[CLOSED] Closed\"," +
+                                            "\"index_uuid\":\"_na_\"," +
+                                            "\"shard\":\"1\"," +
+                                            "\"index\":\"foo\"" +
+                                "}" +
+                            "}," +
+                            "{" +
+                                "\"shard\":2," +
+                                "\"index\":\"foo\"," +
+                                "\"node\":\"node_3\"," +
+                                "\"reason\":{" +
+                                            "\"type\":\"parsing_exception\"," +
+                                            "\"reason\":\"foobar\"," +
+                                            "\"line\":5," +
+                                            "\"col\":7" +
+                                "}" +
+                            "}" +
+                    "]}", builder.string());
+        }
+    }
+
+    public void testToAndFromXContent() throws IOException {
+        final XContent xContent = randomFrom(XContentType.values()).xContent();
+
+        ShardSearchFailure[] shardSearchFailures = new ShardSearchFailure[randomIntBetween(1, 5)];
+        for (int i = 0; i < shardSearchFailures.length; i++) {
+            Exception cause = randomFrom(
+                    new ParsingException(1, 2, "foobar", null),
+                    new InvalidIndexTemplateException("foo", "bar"),
+                    new TimestampParsingException("foo", null),
+                    new NullPointerException()
+            );
+            shardSearchFailures[i] = new  ShardSearchFailure(cause, new SearchShardTarget("node_" + i, new Index("test", "_na_"), i));
+        }
+
+        final String phase = randomFrom("query", "search", "other");
+        SearchPhaseExecutionException actual = new SearchPhaseExecutionException(phase, "unexpected failures", shardSearchFailures);
+
+        BytesReference exceptionBytes = XContentHelper.toXContent(actual, xContent.type(), randomBoolean());
+
+        ElasticsearchException parsedException;
+        try (XContentParser parser = createParser(xContent, exceptionBytes)) {
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            parsedException = ElasticsearchException.fromXContent(parser);
+            assertEquals(XContentParser.Token.END_OBJECT, parser.currentToken());
+            assertNull(parser.nextToken());
+        }
+
+        assertNotNull(parsedException);
+        assertThat(parsedException.getHeaderKeys(), hasSize(0));
+        assertThat(parsedException.getMetadataKeys(), hasSize(2));
+        assertThat(parsedException.getMetadata("es.phase"), hasItem(phase));
```

Comment:
the es. prefix here is a bit weird, but I don't know how to remove it without breaking bw comp. We have to take into account though that even if the info is available, no users will know how to retrieve it :)